src/DataSource/AzureDevOpsPullRequestSource.cs

```diff
@@ -158,52 +153,51 @@ await foreach (var pr in FetchAccountActivePullRequsts(client, userId, accountCo
                 {
                     //  Skip this review if we aren't assigned.
                     //
-                    continue;
+                    return null;
                 }
 
                 // Skip declined reviews.
                 //
                 if (reviewer.HasDeclined.HasValue && reviewer.HasDeclined.Value)
                 {
-                    continue;
+                    return null;
                 }
 
                 // If we have already casted a "final" vote, then skip it.
                 //
                 if (reviewer.HasFinalVoteBeenCast())
                 {
-                    m_statistics.SignedOff++;
-
-                    if (state == PrState.SignedOff)
-                    {
-                        yield return pr;
-                    }
-
-                    continue;
+                    return PrState.SignedOff;
                 }
 
-                // TODO: It would be nice if there was a way to tell if
-                // the review was changed since you started waiting.
-                //
                 if (reviewer.IsWaiting())
                 {
-                    m_statistics.Waiting++;
+                    // If we are waiting on the PR, inspect the active threads in the PR.
+                    // If we have left a comment in a thread that is still active, the PR is not actionable to us.
+                    // If we there are no active threads where we have participated, the PR is actionable to us.
+                    //
+                    List<GitPullRequestCommentThread> threads = await client.GetThreadsAsync(pr.Repository.Id, pr.PullRequestId);
 
-                    if (state == PrState.Waiting)
-                    {
-                        yield return pr;
-                    }
+                    bool threadInvolvesUs(GitPullRequestCommentThread thread) => thread.Comments.Any(c => Guid.Parse(c.Author.Id) == userId);
+                    bool anyActiveThreads = threads.Any(t => t.Status == CommentThreadStatus.Active && threadInvolvesUs(t));
```

Comment:
Should we include Pending comments as well?

Reply:
Yes